specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,267 @@
+  /// Programmatically trigger a save as action for current content.
+  ///
+  /// Opens a system modal dialog by default. Returns COREWEBVIEW2_SAVE_AS_OPEN_SYSTEM_DIALOG.
+  /// If it was already opened, this method would not open another one
+  ///
+  /// If the silent save as option is enabled, won't open the system dialog, will
+  /// raise the `SaveAsRequested` event instead and process through its event args. The method can return
+  /// a detailed info to indicate the call's result. Please see COREWEBVIEW2_SAVE_CONTENT_AS_RESULTS
+  ///
+  /// \snippet ScenarioSaveAs.cpp ProgrammaticSaveAs
+  HRESULT SaveContentAs([in] ICoreWebView2StagingSaveContentAsCompletedHandler* handler);
+
+  /// Add an event handler for the `SaveAsRequested` event. This event is raised
+  /// when save as is triggered, programmatically or manually.
+  ///
+  /// \snippet ScenarioSaveAs.cpp ToggleSilent
+  HRESULT add_SaveAsRequested(
+   [in] ICoreWebView2StagingSaveAsRequestedEventHandler* eventHanlder,
+   [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_SaveAsRequested`.
+  ///
+  /// \snippet ScenarioSaveAs.cpp ToggleSilent
+  HRESULT remove_SaveAsRequested(
+   [in] EventRegistrationToken token);
+}
+
+/// The event handler for the `SaveAsRequested` event, when the handler
+/// exists, the silent save as enables.
+[uuid(55b86cd2-adfd-47f1-9cef-cdfb8c414ed3), object, pointer_default(unique)]
+interface ICoreWebView2StagingSaveAsRequestedEventHandler : IUnknown {
+  HRESULT Invoke(
+   [in] ICoreWebView2* sender,
+   [in] ICoreWebView2StagingSaveAsRequestedEventArgs* args);
+}
+
+/// The event args for `SaveAsRequested` event
+[uuid(80101027-b8c3-49a1-a052-9ea4bd63ba47), object, pointer_default(unique)]
+interface ICoreWebView2StagingSaveAsRequestedEventArgs : IUnknown {
+  /// Get the Mime type of content to be saved
+  [propget] HRESULT ContentMimeType([out, retval] LPWSTR* value);
+
+  /// Indicates if this event is a silent save as job. `Handled` as FALSE means
+  /// save as handled by system default dialog; TRUE means a silent save as,
+  /// will skip the system dialog.
+  ///
+  /// Set the `Handled` for save as
+  [propput] HRESULT Handled ([in] BOOL handled);
+
+  /// Get the `Handled` for save as
+  [propget] HRESULT Handled ([out, retval] BOOL* handled);
+
+  /// Indicates if a silent save as confirm to download, TRUE means confirm.
+  /// when the event is invoked, the download will start. A programmatic call will
+  /// return COREWEBVIEW2_SAVE_AS_STARTED as well. set it FASLE to cancel save as 
+  /// and will return COREWEBVIEW2_SAVE_AS_CANCELLED.
+  ///
+  /// Set the `ConfrimToSave` for save as
+  [propput] HRESULT ConfirmToSave ([in] BOOL confirmToSave);
+
+  /// Get the `ConfrimToSave` for save as
+  [propget] HRESULT ConfirmToSave ([out, retval] BOOL* confirmToSave);
+
+  /// Returns an `ICoreWebView2Deferral` object. Use this operation to complete
```

Comment:
We need to say in the documentation what this defers.

```suggestion
  /// Returns an `ICoreWebView2Deferral` object. This will defer showing the default Save As dialog and performing the Save As operation.
```